src/chunk_constraint.c

```diff
@@ -467,6 +482,46 @@ get_slice_with_id(const Hypercube *cube, int32 id)
 	return NULL;
 }
 
+static void
+check_chunk_violate_constraint(Oid chunk_relid, const Dimension *dim, const DimensionSlice *slice)
+{
+	Relation rel;
+	TupleTableSlot *slot;
+	TableScanDesc scandesc;
+	bool isnull;
+
+	rel = table_open(chunk_relid, AccessShareLock);
+	scandesc = table_beginscan(rel, GetLatestSnapshot(), 0, NULL);
+	slot = table_slot_create(rel, NULL);
+
+	while (table_scan_getnextslot(scandesc, ForwardScanDirection, slot))
+	{
+		Datum datum;
+		int64 value;
+		if (NULL != dim->partitioning)
```

Comment:
Any dimension can have a partitioning func, even time dimensions (e.g., to map a non-time value to a time value).

You need to first apply the function, if it exists, then convert to internal time if it is a time dimension. 

See `ts_hyperspace_calculate_point` for an example, or just call that function.